### DISCUSSION TITLE
Potential API limitations

### DISCUSSION THREAD

**Discussion by rhld16** [sawyerpollard/Minecraft-Weather-New-Tab#19] [Q&A] [answered]
Just wanted to touch upon how the success of this extension could be it's downfall; openweathermap's free API plan allows for 1 million calls per month. If we take the Chrome Store page's "7000+ users" that means every individual user can have the browser open for about 71 hours a month. While this may be far from a problem as of now, continued growth of the extension could mean the API limit being hit way too early.

What would be the best way to stop this from happening?

**Comment by GreatSymphonia** [comment]
Actually, a (bad) solution would be to redirect the users to the openweather so they can create their own API key.
The other alternative would be to increase the delay at which the extension refreshes its data.
The last one would be to pay...

**Comment by vulpesfelix** [comment]
I think OpenWeatherMap raises the API calls cap for open source projects

**Comment by sawyerpollard** [comment, marked as answer]
@rhld16 @Frantee Thanks for the concerns! For now, I just am on a student plan that provides 3000 API calls per minute. Currently, the extension fetches new weather from the API every 30 minutes. So, I believe the extension could handle 90,000 active users at once. If that math makes sense.

**Reply by rhld16** [reply]
> @rhld16 @Frantee Thanks for the concerns! For now, I just am on a student plan that provides 3000 API calls per minute. Currently, the extension fetches new weather from the API every 30 minutes. So, I believe the extension could handle 90,000 active users at once. If that math makes sense.

And that's assuming that the browser is open with everyone 24/7 so that number could, in fact, be anything from 2x to 16x higher depending on each users' activity. Sounds like a reasonable solution for now anyway hah.
